modules/ossm-accessing-bookinfo-application-using-istio-gateway-injection.adoc

```diff
@@ -0,0 +1,44 @@
+:_mod-docs-content-type: Procedure
+[id="ossm-accessing-bookinfo-application-using-istio-gateway-injection"]
+= Accessing the Bookinfo application by using Istio gateway injection
+:context: ossm-accessing-bookinfo-application-using-istio-gateway-injection
+
+Gateway injection uses the same mechanisms as {istio} sidecar injection to create a gateway from a `Deployment` resource that is paired with a `Service` resource. The `Service` resource can be made accessible from outside an {ocp-product-title} cluster.
+
+.Procedure
+
+. Create the `istio-ingressgateway` deployment and service by running the following command at the CLI:
++
+[source,terminal]
+----
+$ oc apply -n istio-system -f ingress-gateway.yaml
+----
+
+. Configure the `bookinfo` application to use the new gateway by applying the gateway configuration: 
++
+[source,terminal]
+----
+$ oc apply -f https://raw.githubusercontent.com/istio/istio/master/samples/bookinfo/networking/bookinfo-gateway.yaml -n bookinfo
+----
++
+[NOTE]
+====
+To configure gateway injection with the `bookinfo` application, this example uses a sample gateway configuration file that must be applied in the namespace where the application is installed.
+====
+
+. Use a route to expose the gateway external to the cluster:
++
+[source,terminal]
+----
+$ oc expose service istio-ingressgateway -n istio-system
+----
+
```

Comment:
This will likely need a wait command as well to wait for the route to be fully setup although I don't know what that is off the top of my head for openshift.

Reply:
I discussed this with Marko early this morning while he was online. He said he doesn't think we need to issue a wait in this context because the host is usually set immediately. 

The wait usually looks at status.conditions but we can't do that for a route.

Any additional technical details are beyond my comprehension at this point ;-)